walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,49 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.PriorityQueue;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    PriorityQueue<Event> eventQueue = new PriorityQueue<>(Event.ORDER_BY_START_TIME);
+    eventQueue.addAll(events);
+
+    Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    int currStartTime = TimeRange.START_OF_DAY;
+
+    while (!eventQueue.isEmpty()) {
+      Event currEvent = eventQueue.remove();
```

Comment:
nit: the `curr` prefix doesn't seem necessary (I don't see another Event object in scope). If you do want to keep it, better to write it out as `currentEvent`, or perhaps give it a slightly more accurate name like `existingEvent`.